persistent/Database/Persist/Class/PersistUnique.hs

```diff
@@ -96,13 +137,59 @@ class (PersistUniqueRead backend, PersistStoreWrite backend) =>
     --
     -- * insert the new record if it does not exist;
     -- * update the existing record that matches the given uniqueness constraint.
+    --
+    -- === __Example usage__
+    --
+    -- We try to explain 'upsertBy' using the schema 2.
+    --
+    -- @
+    -- upsertBy (UniqueUserName \"SPJ\") (Person \"X\" 999) [PersonAge +=. 15]
```

Comment:
If you'd like, you can use the "bird track" haddock code syntax, so you don't have to escape things:

```
> upsertBy (UniqueUserName "SPJ") (Person "X" 999) [PersonAge +=. 15]
```

The main advantage to the `@` syntax is that you can create hyperlinks to identifiers using single quotes, but if you aren't doing that then it's noisier to use `@`